tests/legacy/framework/class-wc-unit-test-case.php

```diff
@@ -150,4 +151,78 @@ public static function file_copy( $source, $dest ) {
 		self::reenable_code_hacker();
 		return $result;
 	}
+
+	/**
+	 * Get an instance of a class that has been registered in the dependency injection container.
+	 * To get an instance of a legacy class (such as the ones in the 'íncludes' directory) use
+	 * 'get_legacy_instance_of' instead.
+	 *
+	 * @param string $class_name The class name to get an instance of.
+	 *
+	 * @return mixed The instance.
+	 */
+	public function get_instance_of( string $class_name ) {
+		return wc_get_container()->get( $class_name );
+	}
+
+	/**
+	 * Get an instance of  legacy class (such as the ones in the 'íncludes' directory).
+	 * To get an instance of a class registered in the dependency injection container use 'get_instance_of' instead.
+	 *
+	 * @param string $class_name The class name to get an instance of.
+	 *
+	 * @return mixed The instance.
+	 */
+	public function get_legacy_instance_of( string $class_name ) {
+		return wc_get_container()->get( LegacyProxy::class )->get_instance_of( $class_name );
+	}
+
+	/**
+	 * Reset all the cached resolutions in the dependency injection container, so any further "get"
+	 * for shared definitions will generate the instance again.
+	 * This may be needed when registering mocks for already resolved shared classes.
+	 */
+	public function reset_container_resolutions() {
```

Comment:
shouldn't these reset methods be hooked tearDown methods for all test classes?

Reply:
Not necessarily. You need that only when you are mocking a shared class.